src/shared/components/app/app.tsx

```diff
@@ -40,17 +40,15 @@ export class App extends Component<any, any> {
             <Navbar siteRes={siteRes} />
             <div className="mt-4 p-0 fl-1">
               <Switch>
-                {routes.map(
-                  ({ path, exact, component: Component, ...rest }) => (
-                    <Route
-                      key={path}
-                      path={path}
-                      exact={exact}
-                      render={props => <Component {...props} {...rest} />}
-                    />
-                  )
-                )}
-                <Route render={props => <NoMatch {...props} />} />
+                {routes.map(({ path, component }) => (
+                  <Route
+                    key={path}
+                    path={path}
+                    exact={true}
```

Comment:
I made them all exact so that the 404 page would show up when expected.

To see an example, try visiting this link: https://lemmy.ml/instances/a/bunch/of/route/parts. I would expect this to show a 404. `/instances` is a valid route, but all of that stuff at the end should mean that the route leads either to another page on the website or a non existing page. As you'll find out from following the link, it still leads to the instances page. Making the routes exact prevents this issue.